Add scalar conversion functions

davidnevadoc opened the pull request:
Resolves #26

davidnevadoc committed:
Add scalar conversion functions

iquerejeta commented on src/util.rs:
```diff
@@ -77,17 +77,63 @@ pub fn get_domain_attrs<F: FftField>(
+        assert!( s < modulus,
+            "The embedded scalar exceeds the capacity representation of the outter curve scalar");
+    } else {
+        let m = <<P::ScalarField as PrimeField>::BigInt as BigInteger>::from_bits_le(
+            &modulus.to_bits_le(),
```
Why are we doing this? if `modulus.num_bits() < scalar_repr.num_bits()`, then it should directly fail, right?

davidnevadoc replied:
I was thinking of the following case:
 1. The embedded scalar field  `P::ScalarField` is larger than the pairing friendly curve scalar field `E::Fq` and it needs more bits to be represented. 
 2. The used embedded scalar still 'fits' in `E::Fq` because it is lesser than its module.

iquerejeta replied:
aha! I see. Then it makes sense. I (incorrectly) assumed that `into_repr()` was fitting only to the required bits. All good then 👍

iquerejeta commented on src/util.rs:
```diff
@@ -77,17 +77,63 @@ pub fn get_domain_attrs<F: FftField>(
+        assert!( s < modulus,
+            "The embedded scalar exceeds the capacity representation of the outter curve scalar");
+    } else {
+        let m = <<E::Fr as PrimeField>::BigInt as BigInteger>::from_bits_le(
+            &modulus.to_bits_le(),
+        );
```
Same

iquerejeta reviewed (approved):
Looks good to me. However, if these functions are only used for tests, I would change the visibility of them. And, whenever there is the expectation of using these elsewhere, I would give a thorough description of which cases these functions must be used.

davidnevadoc committed:
Change function visibility

Change visiblity for `from_embedded_curve_scalar` and
`to_embedded_curve_scalar`
